storage/remote/client.go

```diff
@@ -287,6 +306,10 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	if len(resp.Results) != len(req.Queries) {
 		return nil, errors.Errorf("responses: want %d, got %d", len(req.Queries), len(resp.Results))
 	}
+	return FromQueryResult(sortSeries, resp.Results[0]), nil
+}
 
-	return resp.Results[0], nil
+func (c *Client) handleStreamedResponse(httpResp *http.Response) (storage.SeriesSet, error) {
+	stream := NewChunkedReader(httpResp.Body, DefaultChunkedReadLimit, nil)
```

Comment:
This method might be too shallow I would just inline this.

Reply:
Done.